core/DCache.cpp

```diff
@@ -20,12 +20,16 @@ namespace olympia
         in_lsu_lookup_req_.registerConsumerHandler(
             CREATE_SPARTA_HANDLER_WITH_DATA(DCache, receiveMemReqFromLSU_, MemoryAccessInfoPtr));
 
+        in_l2cache_resp_.registerConsumerHandler(
+            CREATE_SPARTA_HANDLER_WITH_DATA(DCache, receiveRespFromL2Cache_, MemoryAccessInfoPtr));
+
         in_l2cache_ack_.registerConsumerHandler(
             CREATE_SPARTA_HANDLER_WITH_DATA(DCache, receiveAckFromL2Cache_, uint32_t));
 
-        in_l2cache_resp_.registerConsumerHandler(
-            CREATE_SPARTA_HANDLER_WITH_DATA(DCache, receiveRespFromL2Cache_, MemoryAccessInfoPtr));
+        in_l2cache_resp_.registerConsumerEvent(in_l2_cache_resp_receive_event_);
+        in_lsu_lookup_req_.registerConsumerEvent(in_lsu_lookup_req_receive_event_);
```

Comment:
@knute-mips I was not 100% if this is what you meant in the last call.

Reply:
Yes!  Actually it is.  Sorry, meant to give you an example of it.  Glad you figured it out!